### PR TITLE
Service-Portal : National registry translations

### PR DESCRIPTION
**Pull request by obmagnusson**
# \<Description\>

Attach a link to issue if relevant

## What

Specify what you're trying to achieve

## Why

Specify why you need to achieve this

## Screenshots / Gifs

Attach Screenshots / Gifs to help reviewers understand the scope of the pull request

## Checklist:

- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] Formatting passes locally with my changes
- [x] I have rebased against main before asking for a review

### CODE REVIEW

File: libs/api/domains/national-registry/src/lib/soap/nationalRegistryApi.ts [hunk 1]
```diff
@@ -13,6 +13,9 @@ import {
   GetViewFamilyDto,
 } from './dto'
 import { FamilyMember, User, FamilyRelation } from '../types'
+import { Gender } from '../types/gender.enum'
+import { MaritalStatus } from '../types/maritalStatus.enum'
```
Comment (darrikonn) [inline libs/api/domains/national-registry/src/lib/soap/nationalRegistryApi.ts:17]: Tidy the imports by importing from the index file (like the import above from `../types`.

File: libs/api/domains/national-registry/src/lib/models/familyMember.model.ts [hunk 2]
```diff
@@ -10,11 +13,13 @@ export class NationalRegistryFamilyMember {
   @Field(() => String)
   fullName!: string
 
-  @Field(() => String)
-  gender!: string
+  @IsEnum(Gender)
```
Comment (darrikonn) [inline libs/api/domains/national-registry/src/lib/models/familyMember.model.ts:16]: This is combining dtos and GraphQL types.
Don't need the class-validator on graphql objects.

File: libs/api/domains/national-registry/src/lib/models/familyMember.model.ts [hunk 3]
```diff
@@ -1,6 +1,9 @@
 import { Field, ObjectType, ID } from '@nestjs/graphql'
+import { IsEnum } from 'class-validator'
 
 import { FamilyRelation } from '../types/familyRelation.enum'
+import { Gender } from '../types/gender.enum'
+import { MaritalStatus } from '../types/maritalStatus.enum'
```
Comment (darrikonn) [inline libs/api/domains/national-registry/src/lib/models/familyMember.model.ts:6]: ditto with tidying up imports.

File: libs/api/domains/national-registry/src/lib/models/user.model.ts [hunk 4]
```diff
@@ -1,4 +1,16 @@
...
+import { MaritalStatus } from '../types/maritalStatus.enum'
+import { BanMarking } from './banMarking.model'
+
+registerEnumType(MaritalStatus, {
+  name: 'MaritalStatus',
+})
+
+registerEnumType(Gender, {
+  name: 'Gender',
+})
```
Comment (darrikonn) [inline libs/api/domains/national-registry/src/lib/models/user.model.ts:13]: Not sure if it's correct to register widely used enums in the `user.model` file

File: libs/api/domains/national-registry/src/lib/models/user.model.ts [hunk 5]
```diff
@@ -1,4 +1,16 @@
-import { Field, ObjectType, ID } from '@nestjs/graphql'
+import { Field, ObjectType, ID, registerEnumType } from '@nestjs/graphql'
+import { IsEnum } from 'class-validator'
```
Comment (darrikonn) [inline libs/api/domains/national-registry/src/lib/models/user.model.ts:2]: Ditto for class-validator

File: libs/api/domains/national-registry/src/lib/nationalRegistry.service.ts [hunk 6]
```diff
@@ -1,9 +1,7 @@
 import { Injectable } from '@nestjs/common'
-import { ApolloError } from 'apollo-server-express'
-
-import { logger } from '@island.is/logging'
 import { FamilyMember, User } from './types'
 import { NationalRegistryApi } from './soap/nationalRegistryApi'
+import { BanMarking } from './models/banMarking.model'
```
Comment (darrikonn) [inline libs/api/domains/national-registry/src/lib/nationalRegistry.service.ts:4]: Should not be importing a GraphQL model into the service.

File: libs/api/domains/national-registry/src/lib/user.resolver.ts [hunk 7]
```diff
@@ -10,6 +10,7 @@ import {
 import { NationalRegistryUser } from './models'
 import { NationalRegistryService } from './nationalRegistry.service'
 import { User } from './types'
+import { BanMarking } from './models/banMarking.model'
```
Comment (darrikonn) [inline libs/api/domains/national-registry/src/lib/user.resolver.ts:13]: Models already imported above.

**Review by darrikonn — CHANGES_REQUESTED** [review]

File: libs/api/domains/national-registry/src/lib/models/user.model.ts (hunk 4 above)
Comment (obmagnusson) [reply libs/api/domains/national-registry/src/lib/models/user.model.ts:11]: Yeah agree. Got any suggestion where?

Comment (darrikonn) [reply libs/api/domains/national-registry/src/lib/models/user.model.ts:11]: Most likely with some interaction with the `index` file... either the enums should be registered there, or they should be defined in `scalars.ts` file (whould choose that option) and with an `export * from './scalars'` in the index file.

**Review by darrikonn — APPROVED** [review]
Nice

**Review by Thorrik58 — APPROVED** [review]

**Pull request merged**